ddtrace/commands/ddtrace_run.py

```diff
@@ -16,54 +27,89 @@
+Execute the given Python command after configuring it to emit Datadog traces.
+The Datadog profiler can also be enabled.
 
-Usage: ddtrace-run <my_program>
-"""  # noqa: E501
 
-
-def _ddtrace_root():
-    from ddtrace import __file__
-    return os.path.dirname(__file__)
+Examples
+ddtrace-run python app.py
+ddtrace-run uwsgi app.py
+ddtrace-run gunicorn myproject.wsgi
+"""
 
 
 def _add_bootstrap_to_pythonpath(bootstrap_dir):
     """
     Add our bootstrap directory to the head of $PYTHONPATH to ensure
     it is loaded before program code
     """
-    python_path = os.environ.get('PYTHONPATH', '')
+    python_path = os.environ.get("PYTHONPATH", "")
 
     if python_path:
-        new_path = '%s%s%s' % (bootstrap_dir, os.path.pathsep, os.environ['PYTHONPATH'])
-        os.environ['PYTHONPATH'] = new_path
+        new_path = "%s%s%s" % (bootstrap_dir, os.path.pathsep, os.environ["PYTHONPATH"])
+        os.environ["PYTHONPATH"] = new_path
     else:
-        os.environ['PYTHONPATH'] = bootstrap_dir
+        os.environ["PYTHONPATH"] = bootstrap_dir
 
 
 def main():
-    if len(sys.argv) < 2 or sys.argv[1] == '-h':
-        print(USAGE)
-        return
-
-    log.debug('sys.argv: %s', sys.argv)
-
-    root_dir = _ddtrace_root()
-    log.debug('ddtrace root: %s', root_dir)
-
-    bootstrap_dir = os.path.join(root_dir, 'bootstrap')
-    log.debug('ddtrace bootstrap: %s', bootstrap_dir)
+    parser = argparse.ArgumentParser(
+        description=USAGE,
+        prog="ddtrace-run",
+        usage="ddtrace-run <your usual python command>",
+        formatter_class=argparse.RawTextHelpFormatter,
+    )
+    parser.add_argument("command", nargs=argparse.REMAINDER, type=str, help="Command string to execute.")
+    parser.add_argument("-d", "--debug", help="enable debug mode (disabled by default)", action="store_true")
+    parser.add_argument("-i", "--info", help="print library info useful for debugging", action="store_true")
+    parser.add_argument("-p", "--profiling", help="enable profiling (disabled by default)", action="store_true")
+    parser.add_argument("-v", "--version", action="version", version="%(prog)s " + ddtrace.__version__)
+    args = parser.parse_args()
+
+    if args.profiling:
+        os.environ["DD_PROFILING_ENABLED"] = "true"
+
+    if args.debug:
+        os.environ["DD_TRACE_DEBUG"] = "true"
+
+    if args.info:
+        # Inline imports for performance.
+        import pprint
+        from ddtrace.internal.debug import collect
+
+        pprint.pprint(collect(ddtrace.tracer))
+        sys.exit(0)
+
+    root_dir = os.path.dirname(ddtrace.__file__)
+    log.debug("ddtrace root: %s", root_dir)
+
+    bootstrap_dir = os.path.join(root_dir, "bootstrap")
+    log.debug("ddtrace bootstrap: %s", bootstrap_dir)
 
     _add_bootstrap_to_pythonpath(bootstrap_dir)
-    log.debug('PYTHONPATH: %s', os.environ['PYTHONPATH'])
-    log.debug('sys.path: %s', sys.path)
+    log.debug("PYTHONPATH: %s", os.environ["PYTHONPATH"])
```

Comment:
python version? maybe python executable location?

Reply:
If debug logs are enabled then we're going to have the tracer debug logs printed out which includes the python version and much more. The python executable is already logged later on below.